### ISSUE TITLE
FP identity_op does not catch type hint is needed

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#14932]
### Summary

.

### Lint Name

identity_op

### Reproducer

I tried this code:

```rust
fn main() {
    0usize + &Default::default();
}
```

I saw this happen:

```
warning: this operation has no effect
 --> src/main.rs:2:5
  |
2 |     0usize + &Default::default();
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: consider reducing it to: `Default::default()`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#identity_op
  = note: `#[warn(clippy::identity_op)]` on by default
```

however the suggested code no longer typechecks 😆 

```rust
fn main() {
    Default::default();
}

```
error[E0790]: cannot call associated function on trait without specifying the corresponding `impl` type
 --> src/main.rs:2:5
  |
2 |     Default::default();
  |     ^^^^^^^^^^^^^^^^^^ cannot call associated function of trait
  |
help: use a fully-qualified path to a specific available implementation
  |
2 |     </* self type */ as Default>::default();
  |     +++++++++++++++++++        +

error: aborting due to 1 previous error
```

### Version

```text

```
